core/src/main/java/de/bwaldvogel/mongo/oplog/CollectionBackedOplog.java

```diff
@@ -26,18 +27,40 @@ public void handleInsert(String databaseName, Document query) {
                 .withTimestamp(new BsonTimestamp(instant.toEpochMilli()))
                 .withWall(instant)
                 .withOperationType(OperationType.INSERT)
-                .withOperationDocument(d)
+                .withOperationDocument(d.clone())
                 .withNamespace(String.format("%s.%s", databaseName, query.get("insert")))
                 .asDocument()).collect(Collectors.toList());
         collection.insertDocuments(oplogDocuments);
     }
 
     @Override
-    public void handleUpdate(String databaseName, Document query) {
+    public void handleUpdate(String databaseName, Document query, List<Object> updatedIds) {
+        Instant instant = clock.instant();
+        Document updateDoc = ((List<Document>) query.get("updates")).get(0);
+        List<Document> oplogDocuments = updatedIds.stream().map(id ->
+            new OplogDocument()
+                .withTimestamp(new BsonTimestamp(instant.toEpochMilli()))
+                .withWall(instant)
+                .withOperationType(OperationType.UPDATE)
+                .withOperationDocument(buildUpdateOperationDocument(updateDoc))
```

Comment:
do we also need `updateDescription.updatedFields` and `updateDescription.removedFields`?